bam_sort.c

```diff
@@ -1166,13 +1167,35 @@ int bam_merge_core2(int by_qname, const char *out, const char *mode,
     if (!iter) goto mem_fail;
     hdr = (bam_hdr_t**)calloc(n, sizeof(bam_hdr_t*));
     if (!hdr) goto mem_fail;
+
+    if (flag & MERGE_NO_TRANS) {
+        if (hout) bam_hdr_destroy(hout);
+        hout = NULL;
+        for (i = 0; i < n; ++i) {
+            fp[i] = sam_open_format(fn[i], "r", in_fmt);
+            if (fp[i] == NULL) {
+                print_error_errno(cmd, "fail to open \"%s\"", fn[i]);
+                goto fail;
+            }
+            hin = sam_hdr_read(fp[i]);
+            if (!hout) hout = hin;
+            if (hts_get_format(fp[i])->format == sam) hdr[i] = hin;
+            else { if (hin != hout) bam_hdr_destroy(hin); hdr[i] = NULL; }
+        }
+        hin = NULL;
+        // Also skips region parsing; valid assumption.
+        // When the FIXME regarding iteration of SAM files is fixed we'll
+        // be able to make this a tidier goto: after the if (reg) blocks.
+        goto no_trans_tbl;
+    }
+
     translation_tbl = (trans_tbl_t*)calloc(n, sizeof(trans_tbl_t));
     if (!translation_tbl) goto mem_fail;
     RG = (char**)calloc(n, sizeof(char*));
     if (!RG) goto mem_fail;
 
     // prepare RG tag from file names
-    if (flag & MERGE_RG) {
+    if ((flag & MERGE_RG)) {
```

Comment:
No idea to be honest; it doesn't ring any bells, but it's been a while.   It could be removed and then see if warnings do indeed appear.